File: ImagensApp/ImagensApp/ImagensApp/Services/GiphyService.cs
using System.Net.Http;
using System.Threading.Tasks;
using Microsoft.Extensions.Configuration;
using Newtonsoft.Json.Linq;

public class GiphyService
{
    private readonly HttpClient _httpClient;
    private readonly string _apiKey;

    public GiphyService(HttpClient httpClient, IConfiguration configuration)
    {
        _httpClient = httpClient;
        _apiKey = configuration["Giphy:ApiKey"];
    }

    public async Task<JObject> SearchGifsAsync(string query, int limit = 20)
    {
        var response = await _httpClient.GetStringAsync($"https://api.giphy.com/v1/gifs/search?api_key={_apiKey}&q={query}&limit={limit}");
        return JObject.Parse(response);
    }
}

File: ImagensApp/Frontend/frontend/src/Services/GiphyService.js
import axios from 'axios';

const GIPHY_API_KEY = '0j8PaMDFtQmKu6VupE3Fu423glu8QmTG'; // Replace with your actual Giphy API key
const GIPHY_API_URL = 'https://api.giphy.com/v1/gifs/search';

const searchGifs = async (query, limit = 20) => {
  const response = await axios.get(GIPHY_API_URL, {
    params: {
      api_key: GIPHY_API_KEY,
      q: query,
      limit: limit
    }
  });
  return response.data;
};

export default { searchGifs };
